linalg/vector.cpp

```diff
@@ -135,11 +135,13 @@ Vector &Vector::operator=(const Vector &v)
    UseDevice(v.UseDevice());
 #else
    SetSize(v.Size());
-   const bool use_dev = UseDevice() || v.UseDevice();
+   bool vuse = v.UseDevice();
+   const bool use_dev = UseDevice() || vuse;
    v.UseDevice(use_dev);
    // keep 'data' where it is, unless 'use_dev' is true
    if (use_dev) { Write(); }
    data.CopyFrom(v.data, v.Size());
+   v.UseDevice(vuse);
```

Comment:
Why did you needed to split this `UseDev`/`CopyFrom`?

Reply:
Because of `const` correctness, and I believe memory flags should respect that. In this case, the `MemoryFlags` of v are restored back. The current memory manager does not properly handle Synched memory (through aliases). if you remove this line ex11p will error out using a debug version of PETSc.